tensorflow/go/session.go

```diff
@@ -65,6 +65,53 @@ func NewSession(graph *Graph, options *SessionOptions) (*Session, error) {
 	return s, nil
 }
 
+type Device struct {
+	Name, Type  string
+	MemoryLimit int64
+}
+
+// Return list of devices associated with a Session
+func (s *Session) ListDevices() ([]Device, error) {
+	var devices []Device
+
+	status := newStatus()
+	devices_list := C.TF_SessionListDevices(s.c, status.c)
+	if err := status.Err(); err != nil {
+		return nil, fmt.Errorf("TF_SessionListDevices() failed: %v", err)
```

Comment:
Listing the C API method in the error string might be confusing to a Go user since most users should have no idea what `TF_Session*` means.

Perhaps rephrase the error messages? For example, this one can perhaps just be `return nil, err`, while the ones in the loop can be `return nil, fmt.Errorf("failed to fetch name for device #%d: %v", i, err)` etc.?

Reply:
A rationale behind this type of error message texts is simple. By copying exact failed function name to an error message we allow easier code search in case of potential problems. If we leave a less distinctive text like "failed to fetch name ..." then future developers potentially might have found it harder to find exact place in the code by searching through all similar messages. And yes, these error messages are for developers eyes mostly, because normally these functions do not fail.